### PR TITLE
Initialise repo with golang boilerplate

### PR DESCRIPTION
**Pull request by juliaogris**
Initialise tooling with hermit by running

      hermit init
      hermit install make go golangci-lint goreleaser

Initialise go project with

   go mod init github.com/juliaogrs/narwhal

Add the usual golang boilerplate for our typical project setup with
Makefile, GitHub Action config, .gitignore.

Auto-release semver patch version on Merge to master on GitHub action.

Use kong for flag parsing and testify for testing.

### CODE REVIEW

File: bin/go-main-stubs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+#
+#
+# This script generates stub scripts for every Go "main" package in this
```
Comment (camh-) [inline bin/go-main-stubs:4]: What are "stub scripts"? Is this script part of hermit? I'm not sure why this is here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	@go tool cover -func=$(COVERFILE) | $(CHECK_COVERAGE) || $(FAIL_COVERAGE)
+
+cover: test  ## Show test coverage in your browser
+	go tool cover -html=$(COVERFILE)
+
+
+CHECK_COVERAGE = awk -F '[ \t%]+' '/^total:/ {print; if ($$3 < $(COVERAGE)) exit 1}'
+FAIL_COVERAGE = { echo '$(COLOUR_RED)FAIL - Coverage below $(COVERAGE)%$(COLOUR_NORMAL)'; exit 1; }
+
+.PHONY: check-coverage check-uptodate cover test
```
Comment (camh-) [inline Makefile:48]: There is no `check-uptodate` target.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+# --- Utilities ----------------------------------------------------------------
+COLOUR_NORMAL = $(shell tput sgr0 2>/dev/null)
+COLOUR_RED    = $(shell tput setaf 1 2>/dev/null)
+COLOUR_GREEN  = $(shell tput setaf 2 2>/dev/null)
+COLOUR_WHITE  = $(shell tput setaf 7 2>/dev/null)
+
+help:
+	@awk -F ':.*## ' 'NF == 2 && $$1 ~ /^[A-Za-z0-9%_-]+$$/ { printf "$(COLOUR_WHITE)%-25s$(COLOUR_NORMAL)%s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
+
+tools:  ## Install build tools not available in hermit in ./out/bin
```
Comment (camh-) [inline Makefile:76]: I feel like this belongs higher up. The rest of this "utilities" section is unmodified. This sort of hides here.

Also, do we really need protoc generators?

File: main_test.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRun(t *testing.T) {
+	require.Nil(t, run(nil))
```
Comment (camh-) [inline main_test.go:10]: Do you prefer `require.Nil` over `require.NoError` ?

Comment (juliaogris) [reply main_test.go:10]: yes, agreed. fixed.

File: Makefile (hunk 3 above)
Comment (juliaogris) [reply Makefile:76]: oh ehm yes. no protoc tools needed. fixed.

File: Makefile (hunk 2 above)
Comment (juliaogris) [reply Makefile:48]: oups! fixed.

File: bin/go-main-stubs (hunk 1 above)
Comment (juliaogris) [reply bin/go-main-stubs:4]: yeah :( not sure what this does. not mine. came with hermit install go i think.

File: Makefile (hunk 2 above)
Comment (camh-) [reply Makefile:48]: Didn't add/push?

**Review by camh- — APPROVED** [review]
:green_apple:

**Pull request merged**